util/model_initializer.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+"""
+This module saves and imports matrices.
+"""
+import numpy
+import os
+
+
+class ModelInitializer(object):
+    """
+    A class for importing and saving models.
+    """
+    def __init__(self, config, n_iterations, verbose=False):
+        """
+        Constructor for model initializer.
+
+        :param dict config: hyperparameters of the recommender, contains _lambda and n_factors.
+        :param int n_iterations: Number of iterations used to train.
+        """
+        self.config = config
+        self.folder = 'matrices'
+        self.config['n_iterations'] = n_iterations
+        self._v = verbose
+
+    def set_config(self, config, n_iterations):
+        """
+        Function that sets config and n_iterations for the initializer.
+
+        :param dict config: hyperparameters of the recommender, contains _lambda and n_factors.
+        :param int n_iterations: Number of iterations used to train.
+        """
+        self.config = config
+        self.config['n_iterations'] = n_iterations
+
+    def save_matrix(self, matrix, matrix_name):
+        """
+        Function that dumps the matrix to a .dat file.
+
+        :param ndarray matrix: Matrix to be dumped.
+        :param str matrix_name: Name of the matrix to be dumped.
+        """
+        path = self._create_path(matrix_name, matrix.shape[0])
+        matrix.dump(path)
+        if self._v:
+            print("dumped to %s" % path)
+
+    def load_matrix(self, config, matrix_name, matrix_shape):
```

Comment:
I think the bug is here, that you are not using self.config (because the one passed from the recommenders misses the n_iterations)